models/DataObject/ClassDefinition/Data.php

```diff
@@ -223,19 +238,25 @@ public function getNoteditable(): bool
         return $this->noteditable;
     }
 
+    /**
+     * @return $this
+     */
     public function setNoteditable(bool $noteditable): static
     {
         $this->noteditable = $noteditable;
 
         return $this;
     }
 
-    public function getIndex(): ?int
+    public function getIndex(): bool
     {
         return $this->index;
     }
 
-    public function setIndex(?int $index): static
+    /**
+     * @return $this
+     */
+    public function setIndex(bool $index): static
```

Comment:
This change also works for updates within Pimcore 11.0 because the file is not in strict mode. The integers are converted to a boolean value.

Reply:
@blankse But this could be a BC break, if we try to set the value of $index to NULL through the API. For eg: $input->setIndex(NULL);. Since it was accepting null values also before this PR.

Reply:
@aryaantony92 The file has no strict mode. So the value should be casted to `false`, or not?

Reply:
@blankse NULL is not casted to boolean in the API call even if the file is not in strict mode.

Reply:
@blankse Think we should keep it nullable also

Reply:
@aryaantony92 I test it. You are right. Only the integers will be casted. I create a followup PR to allow `NULL`

Reply:
@aryaantony92 See https://github.com/pimcore/pimcore/pull/15774

Reply:
@blankse Thanks